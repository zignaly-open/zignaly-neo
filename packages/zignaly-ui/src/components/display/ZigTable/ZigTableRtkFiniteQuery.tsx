import React, { ForwardedRef, forwardRef, useImperativeHandle, useState } from "react";
import { ZigTablePropsFiniteQuery, ZigTableQueryParams, ZigTableQueryRef } from "./types";
import ZigTableData from "./ZigTableData";
import useFinitePaginatedQuery from "./use/useFinitePaginatedQuery";
import { SortingState } from "@tanstack/react-table";

const ZigTableRtkFiniteQuery = <T extends object, V extends ZigTableQueryParams>(
  { useQuery, queryExtraParams, queryDataMapper, ...props }: ZigTablePropsFiniteQuery<T, V>,
  ref: ForwardedRef<ZigTableQueryRef>,
) => {
  const [sorting, setSorting] = useState<SortingState>([]);
  const { pagination, setPagination, isFetching, isLoading, totalPages, data, refetch } =
    useFinitePaginatedQuery({
      useQuery,
      sorting,
      queryExtraParams,
    });

  useImperativeHandle(ref, () => ({
    refetch,
  }));

  const items = (queryDataMapper ? data?.data?.map(queryDataMapper) : data?.data) || [];
  return (
    <ZigTableData
      {...props}
      state={{
        ...(props.state || {}),
        sorting,
      }}
      manualPagination
      data={items as T[]}
      onSortingChange={setSorting}
      pageCount={totalPages}
      loading={isLoading}
      fetching={isFetching}
      pagination={pagination}
      onPaginationChange={setPagination}
    />
  );
};

export default forwardRef(ZigTableRtkFiniteQuery) as <
  T extends object,
  V extends ZigTableQueryParams,
>(
  props: ZigTablePropsFiniteQuery<T, V> & { ref?: ForwardedRef<ZigTableQueryRef> },
) => ReturnType<typeof ZigTableRtkFiniteQuery>;
