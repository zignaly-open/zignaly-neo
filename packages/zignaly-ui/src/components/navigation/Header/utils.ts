// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
