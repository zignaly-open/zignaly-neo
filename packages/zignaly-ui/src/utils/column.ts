import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
