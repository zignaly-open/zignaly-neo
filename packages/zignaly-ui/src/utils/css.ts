import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
