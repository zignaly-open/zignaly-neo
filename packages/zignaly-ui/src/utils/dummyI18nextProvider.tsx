import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
