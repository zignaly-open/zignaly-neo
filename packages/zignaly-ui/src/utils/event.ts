export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
