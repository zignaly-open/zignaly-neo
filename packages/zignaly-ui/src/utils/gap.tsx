import styled from "styled-components";

export const Gap = styled.div<{ gap: number }>`
  ${(props) => `
  padding: ${props.gap}px;
  `}
`;
