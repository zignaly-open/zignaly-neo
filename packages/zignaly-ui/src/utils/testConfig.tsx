import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
