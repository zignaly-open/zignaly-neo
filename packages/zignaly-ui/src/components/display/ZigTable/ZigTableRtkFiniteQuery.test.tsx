import React from "react";
import ZigTableRtkFiniteQuery from "./ZigTableRtkFiniteQuery";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { createColumnHelper } from "../../../index";

const columnHelper = createColumnHelper<{
  id: number;
  name: string;
  symbol: string;
  price: number;
}>();
const columns = [
  columnHelper.accessor("name", {
    header: "Name",
    cell: ({ getValue }) => getValue(),
  }),
  columnHelper.accessor("symbol", {
    header: "Symbol",
    cell: ({ getValue }) => getValue(),
  }),
  columnHelper.accessor("price", {
    header: "Price",
    cell: ({ getValue }) => getValue(),
  }),
];

describe("components/display/ZigTable/ZigTableRtkFiniteQuery", () => {
  it("should render table without crashing", () => {
    const useQuery = jest.fn();
    const { container } = renderWithProvidersUi(
      <ZigTableRtkFiniteQuery useQuery={useQuery} columns={columns} prefixId={"test"} />,
    );
    const tableElement = container.querySelector("#test__table");
    expect(tableElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
