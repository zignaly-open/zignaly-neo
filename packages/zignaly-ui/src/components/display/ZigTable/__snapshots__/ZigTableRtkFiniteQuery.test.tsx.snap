// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigTable/ZigTableRtkFiniteQuery should render table without crashing 1`] = `
<div>
  <div
    class="css-17uq058"
  >
    <table
      class="css-1koyi9s"
      id="test__table"
    >
      <thead>
        <tr>
          <th
            colspan="1"
            id="test-table__header-Name"
          >
            <div
              class="MuiBox-root css-16ho1w4"
            >
              <div
                class="css-1xfidup"
              >
                <div>
                  <span
                    class="MuiTypography-root MuiTypography-body2 css-4qv6if-MuiTypography-root"
                  >
                    Name
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-gxr7hr-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  id="test-table__sorted-icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
          </th>
          <th
            colspan="1"
            id="test-table__header-Symbol"
          >
            <div
              class="MuiBox-root css-16ho1w4"
            >
              <div
                class="css-1xfidup"
              >
                <div>
                  <span
                    class="MuiTypography-root MuiTypography-body2 css-4qv6if-MuiTypography-root"
                  >
                    Symbol
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-gxr7hr-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  id="test-table__sorted-icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
          </th>
          <th
            colspan="1"
            id="test-table__header-Price"
          >
            <div
              class="MuiBox-root css-16ho1w4"
            >
              <div
                class="css-1xfidup"
              >
                <div>
                  <span
                    class="MuiTypography-root MuiTypography-body2 css-4qv6if-MuiTypography-root"
                  >
                    Price
                  </span>
                </div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-gxr7hr-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  id="test-table__sorted-icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
          </th>
          <th
            style="width: 50px;"
          >
            <div
              class="sc-gsnTZi bRCFEw"
              id=""
              role="button"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1htw5nq-MuiButtonBase-root-MuiIconButton-root"
                id="test-table__popover-filter"
                tabindex="0"
                type="button"
              >
                <svg
                  color="#A9A9BA"
                  height="16"
                  width="16"
                />
              </button>
            </div>
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <span
    class="MuiTypography-root MuiTypography-body1 css-5wa3he-MuiTypography-root"
    id="table__empty-message"
  />
  <div
    class="MuiBox-root css-54rtym"
  >
    <div
      class="MuiBox-root css-16qevnq"
    />
    <div
      class="MuiBox-root css-47sj1i"
    >
      <button
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-12p5s93-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        id="test-table__go-zero-page"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FirstPageIcon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
          />
        </svg>
      </button>
      <button
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-12p5s93-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        id="test-table__go-previous-page"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronLeftIcon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </svg>
      </button>
      <div
        class="MuiBox-root css-1h8lykf"
        id="test-table__pages"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 css-1fd2nuh-MuiTypography-root"
        >
          Page
        </span>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-1hbb7p8-MuiTypography-root"
          id="test-table__current-page"
        >
          1
        </h3>
        <span
          class="MuiTypography-root MuiTypography-body1 css-1xdzqt9-MuiTypography-root"
        >
          out of
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-cfpb58-MuiTypography-root"
          id="test-table__total-page-count"
        >
          1
        </span>
      </div>
      <button
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-12p5s93-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        id="test-table__go-next-page"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ChevronRightIcon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </button>
      <button
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-12p5s93-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        id="test-table__go-last-page"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LastPageIcon"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
          />
        </svg>
      </button>
    </div>
    <div
      class="MuiBox-root css-f7f7y6"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 css-1fd2nuh-MuiTypography-root"
      >
        Displaying
      </span>
      <div
        class="css-51rdm7"
      >
        <div
          class="MuiBox-root css-tlwa3f"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class=" css-b62m3t-container"
              id="test-table__items-per-page"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-test-table__items-per-page-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class="zig-react-select__control css-1s2u09g-control"
              >
                <div
                  class="zig-react-select__value-container zig-react-select__value-container--has-value css-dlobye-Component"
                >
                  <div
                    class="zig-react-select__single-value css-tcq528-singleValue"
                  >
                    30
                  </div>
                  <div
                    class="zig-react-select__input-container css-1u4kn00-Component"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-haspopup="true"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class="zig-react-select__input"
                      id="react-select-test-table__items-per-page-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="zig-react-select__indicators css-ny0e4k-Component"
                >
                  <div
                    aria-hidden="true"
                    class="zig-react-select__indicator zig-react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-body1 css-1fd2nuh-MuiTypography-root"
      >
        items
      </span>
    </div>
  </div>
</div>
`;
